src/libsyntax/ext/tt/macro_rules.rs

```diff
@@ -94,7 +94,7 @@ impl<'a> MacResult for ParserAnyMacro<'a> {
             let mut parser = self.parser.borrow_mut();
             match parser.token {
                 token::Eof => break,
-                _ => ret.push(panictry!(parser.parse_impl_item()))
+                _ => ret.push(panictry!(parser.parse_impl_item(false)))
```

Comment:
Someone should double-check this - `make_items` should cover the case fo macro trait impl fns and trait fns, so when `make_impl_items` is called, we can be certain that this is going to be a plain impl without traits, hence this here is false.